pkg/apis/sources/v1alpha1/mongoDbSource_types.go

```diff
@@ -0,0 +1,127 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"knative.dev/pkg/apis"
+	"knative.dev/pkg/apis/duck"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+	"knative.dev/pkg/kmeta"
+	"knative.dev/pkg/webhook/resourcesemantics"
+)
+
+// +genclient
+// +genreconciler
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// MongoDbSource is the Schema for the githubsources API
+// +k8s:openapi-gen=true
+type MongoDbSource struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   MongoDbSourceSpec   `json:"spec"`
+	Status MongoDbSourceStatus `json:"status,omitempty"`
+}
+
+// GetGroupVersionKind returns the GroupVersionKind.
+func (m *MongoDbSource) GetGroupVersionKind() schema.GroupVersionKind {
+	return SchemeGroupVersion.WithKind("MongoDbSource")
+}
+
+var (
+	// Check that MongoDbSource can be validated and defaulted.
+	_ apis.Validatable = (*MongoDbSource)(nil)
+	_ apis.Defaultable = (*MongoDbSource)(nil)
+	// Check that we can create OwnerReferences to a MongoDbSource.
+	_ kmeta.OwnerRefable = (*MongoDbSource)(nil)
+	// Check that MongoDbSource is a runtime.Object.
+	_ runtime.Object = (*MongoDbSource)(nil)
+	// Check that MongoDbSource satisfies resourcesemantics.GenericCRD.
+	_ resourcesemantics.GenericCRD = (*MongoDbSource)(nil)
+	// Check that MongoDbSource implements the Conditions duck type.
+	_ = duck.VerifyType(&MongoDbSource{}, &duckv1.Conditions{})
+	// Check that the type conforms to the duck Knative Resource shape.
+	_ duckv1.KRShaped = (*MongoDbSource)(nil)
+	// MongoDbSourceEventTypes represents the different Event Types
+	MongoDbSourceEventTypes = []string{
+		MongoDbSourceUpdateEventType,
+		MongoDbSourceDeleteEventType,
+	}
+)
+
+const (
+	// MongoDbSourceUpdateEventType is the MongoDbSource CloudEvent type for update.
+	MongoDbSourceUpdateEventType = ""
+
+	// MongoDbSourceDeleteEventType is the MongoDbSource CloudEvent type for deletion.
+	MongoDbSourceDeleteEventType = ""
+)
+
+// MongoDbSourceSpec defines the desired state of MongoDbSource
+type MongoDbSourceSpec struct {
+	// ServiceAccountName holds the name of the Kubernetes service account
+	// as which the underlying K8s resources should be run. If unspecified
+	// this will default to the "default" service account for the namespace
+	// in which the MongoDbSource exists.
+	// +optional
+	ServiceAccountName string `json:"serviceAccountName,omitempty"`
+
+	// MongoDbCredentials is the credential to use to access MongoDb.
+	// Must be a secret. Only Name and Namespace are used.
+	MongoDbCredentials corev1.ObjectReference `json:"credentials,omitempty"`
+
+	// Database is the database to watch for changes
+	Database string `json:"database"`
+
+	// Collection is the collection to watch for changes
+	// +optional
+	Collection string `json:"collection,omitempty"`
+
+	// SourceSpec
+	duckv1.SourceSpec `json:",inline"`
```

Comment:
can you add the comment from the SourceSpec in other places.

Reply:
Not sure what you mean?